src/Symfony/Component/Validator/Constraints/WeekValidator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class WeekValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Week) {
+            throw new UnexpectedTypeException($constraint, Week::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        if (!preg_match('/^\d{4}-W(0[1-9]|[1-4][0-9]|5[0-3])$/', $value)) {
```

Comment:
something we might want to do on all validators:
```suggestion
        if (!preg_match('/^\d{4}-W(0[1-9]|[1-4][0-9]|5[0-3])$/D', $value)) {
```

Reply:
I wasn't aware of this modifier. I added it, I'll have a look at a follow-up PR to add it where it could be relevant. Would it be considered a bugfix or new feature? Not sure on this one